ui/ui-components/pages/app/[id]/edit.tsx

```diff
@@ -47,14 +54,37 @@ export default function Page(props) {
   const [ranTest, setRanTest] = useState(false);
   const { id } = router.query;
 
-  async function edit(event) {
-    event.preventDefault();
+  useEffect(() => {
+    if (!oAuthPopup) return;
+
+    const onMessage = (event: MessageEvent) => {
+      if (event.source === oAuthPopup) {
+        loadOAuthRequest(event.data.requestId);
+      }
+    };
+
+    window.addEventListener("message", onMessage, false);
```

Comment:
I *think* we should be using `globalThis` rather than `window` in these modern times